File: src/main/java/com/aaronhowser1/pitchperfect/enchantments/ModEnchantments.java
package com.aaronhowser1.pitchperfect.enchantments;

import com.aaronhowser1.pitchperfect.InstrumentItem;
import com.aaronhowser1.pitchperfect.PitchPerfect;
import net.minecraft.world.entity.EquipmentSlot;
import net.minecraft.world.item.enchantment.Enchantment;
import net.minecraft.world.item.enchantment.EnchantmentCategory;
import net.minecraftforge.eventbus.api.IEventBus;
import net.minecraftforge.registries.DeferredRegister;
import net.minecraftforge.registries.ForgeRegistries;
import net.minecraftforge.registries.RegistryObject;

public class ModEnchantments {

    public static final EnchantmentCategory INSTRUMENT = EnchantmentCategory.create("INSTRUMENT", item -> item instanceof InstrumentItem);
    public static final DeferredRegister<Enchantment> ENCHANTMENTS =
            DeferredRegister.create(ForgeRegistries.ENCHANTMENTS, PitchPerfect.MOD_ID);

    public static RegistryObject<Enchantment> HEALING_BEAT =
            ENCHANTMENTS.register("healing_beat",
                    () -> new HealingBeatEnchantment(
                            Enchantment.Rarity.COMMON,
                            INSTRUMENT,
                            EquipmentSlot.MAINHAND
                    ));
    public static void register(IEventBus eventBus) {
        ENCHANTMENTS.register(eventBus);

    }
}

File: src/main/kotlin/com/aaronhowser1/pitchperfect/enchantment/ModEnchantments.kt
package com.aaronhowser1.pitchperfect.enchantment

import com.aaronhowser1.pitchperfect.PitchPerfect
import com.aaronhowser1.pitchperfect.item.InstrumentItem
import net.minecraft.world.item.Item
import net.minecraft.world.item.enchantment.EnchantmentCategory
import net.minecraftforge.registries.DeferredRegister
import net.minecraftforge.registries.ForgeRegistries

object ModEnchantments {

    val INSTRUMENT = EnchantmentCategory.create(
        "INSTRUMENT"
    ) { item: Item? -> item is InstrumentItem }

    val ENCHANTMENTS = DeferredRegister.create(ForgeRegistries.ENCHANTMENTS, PitchPerfect.MOD_ID)


}